rangeserver/src/warden_handler.rs

```diff
@@ -124,48 +124,69 @@ impl WardenHandler {
         }
     }
 
-    async fn continuously_connect_and_register(
+    async fn continuously_connect_and_register_inner(
         host_info: HostInfo,
         config: common::config::RegionConfig,
         updates_sender: mpsc::UnboundedSender<WardenUpdate>,
         state: Arc<StartedState>,
         epoch_supplier: Arc<dyn EpochSupplier>,
     ) -> Result<(), WardenErr> {
+        let addr = format!("http://{}", config.warden_address.clone());
```

Comment:
Unrelated but seems too rigid to insist on http scheme.

Reply:
I am under the impression that this is a tonic requirement, is that not the case?

Reply:
I was suggesting we put the scheme in the `config.warden_address` and support https/TLS in the future in the config instead of hardcoding `http` in code.